client/state/data-layer/wpcom-http/test/http.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal dependencies
+ */
+import { WPCOM_HTTP_REQUEST } from 'state/action-types';
+
+/**
+ * Internal dependencies
+ */
+import {
+	apiVersionPattern,
+	apiNamespacePattern,
+	http
+} from '../actions';
+
+describe( 'http', () => {
+	describe( 'apiVersion shape', () => {
+		it( 'should not accept `1` as a valid version', () => {
+			expect( apiVersionPattern.test( '1' ) ).to.be.false;
+		} );
+
+		it( 'should accept `v1` as a valid version', () => {
+			expect( apiVersionPattern.test( 'v1' ) ).to.be.true;
+		} );
+
+		it( 'should not accept `v1.` as a valid version', () => {
+			expect( apiVersionPattern.test( 'v1.' ) ).to.be.false;
+		} );
+
+		it( 'should accept `v1.3` as a valid version', () => {
+			expect( apiVersionPattern.test( 'v1.3' ) ).to.be.true;
+		} );
+	} );
+
+	describe( 'apiNamespace shape', () => {
+		it( 'should not accept `wpcom` as a valid namespace', () => {
+			expect( apiNamespacePattern.test( 'wpcom' ) ).to.be.false;
+		} );
+
+		it( 'should accept `wpcom/v1` as a valid namespace', () => {
+			expect( apiNamespacePattern.test( 'wpcom/v1' ) ).to.be.true;
+		} );
+
+		it( 'should not accept `wpcom/` as a valid namespace', () => {
+			expect( apiNamespacePattern.test( 'wpcom/' ) ).to.be.false;
+		} );
+
+		it( 'should not accept `wpcom/1` as a valid namespace', () => {
+			expect( apiNamespacePattern.test( 'wpcom/1' ) ).to.be.false;
+		} );
+	} );
+
+	describe( 'request action', () => {
+		it( 'should return a valid redux action defining apiVersion', () => {
+			const action = http( {
+				apiVersion: 'v2',
+				method: 'GET',
+				path: '/path/to/endpoint',
+			} );
+
+			expect( action ).to.eql( {
+				type: WPCOM_HTTP_REQUEST,
+				body: {},
+				method: 'GET',
+				path: '/path/to/endpoint',
+				query: {
+					apiVersion: 'v2'
+				},
+				options: {},
+				onSuccess: null,
+				onFailure: null,
+				onProgress: null,
+			} );
```

Comment:
if the test is checking for a valid `apiVersion` then let's not add in a bunch of other tests to it. this is baking fragility into the test by preventing the `http()` function from changing - it's an anti-unit test

instead, we can make the test focus on the testable behavior…

```js
expect( action.query.apiVersion ).to.equal( 'v2' );
```

…and leave everything else out of it